### DISCUSSION TITLE
How do I check if Gruntwork has a module for X technology or service?

### DISCUSSION THREAD

**Discussion by zackproser** [gruntwork-io/knowledge-base#653] [How do I] [answered]
A customer asked: 
> How can I tell if Gruntwork offers a module for a given AWS service or technology?

---

<ins datetime="2023-01-30T16:18:29Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109851">Tracked in ticket #109851</a></p>
</ins>

**Comment by zackproser** [comment, marked as answer]
Thanks for your question. There are a couple of ways to confirm whether or not Gruntwork currently offers a module for a given service or technology: 

# Option 1 - Use our [Repository Browser](https://gruntwork.io/repos/)

![gruntwork-repo-browser](https://user-images.githubusercontent.com/1769996/215873791-7e47595f-a96e-4a2d-8f95-c8643b8e69b7.png)

# Option 2 - use our Infrastructure as Code Library's search page

Try the search bar on [the official page for the Gruntwork Infrastructure as Code Library ](https://gruntwork.io/infrastructure-as-code-library/). If there are no matches for your search query, then it's likely Gruntwork does not currently offer such a module.

![AWS-Infrastructure-as-Code-Library](https://user-images.githubusercontent.com/1769996/215534278-fa6cb144-00e3-4c05-a2f5-5ea6cc18e84c.png)

# Option 3 - Use GitHub's search functionality 

Use GitHub search when logged into GitHub as an account that is a member of the `gruntwork-io` GitHub organization. You can also enter `org:gruntwork-io language:hcl <search-term>` - so, if, for example, you wanted to check for an AWS Neptune module, you could enter `org:gruntwork-io language:hcl neptune` - which would return no results at the time of this writing, indicating no matching module. 

![Search-·-org-gruntwork-io-language-hcl-neptune](https://user-images.githubusercontent.com/1769996/215534729-351b61ae-60f8-40ae-939b-6e988fcc0631.png)

As an alternative example, `org:gruntwork-io language:hcl ecs` does return a number of our modules including our ECS module and our ECS Deploy Runner modules.

![Search-·-org-gruntwork-io-language-hcl-ecs](https://user-images.githubusercontent.com/1769996/215534840-165c77ef-8903-43e4-b1f2-7753be152f33.png)

**Comment by oredavids** [comment]
@zackproser could the [repo-browser](https://gruntwork.io/repos/) be a better first option?

**Reply by zackproser** [reply]
@oredavids Ah yes - almost certainly! Let me amend. Thanks for setting me straight. Didn't realize they were separate.

**Reply by zackproser** [reply]
@oredavids Done - LMK what you think!

**Reply by oredavids** [reply]
Looks good. Going to merge to the docs site.
